### PR TITLE
Improve news scraper

### LINKED ISSUE #1003: Investigate why all Nazareth news items from their RSS feed have been posted again
Let's investigate if it's our fault or the fault of the content of the rss feed.

### PR DESCRIPTION
**Pull request by lucasvanhalst**
Stores rss item guid's instead of url's (when available).
This means when the url changes of the item, no newsitem is (wrongly) created
Also improves performance by reducing amount of db calls

Closes #1003

### CODE REVIEW

File: src/solutions/common/cron/news/rss.py [hunk 1]
```diff
@@ -86,39 +92,37 @@ def _worker(rss_settings_key):
...
 
+        scraper_items = {item.key.id(): item for item in ndb.get_multi(keys) if item}
+
+        if dry_run:
+            rss_link.dry_runned = True
+            must_update_settings = True
+
+        def trans():
+            to_put = []
+            for title, url, guid, description_html, message in items:
```
Comment (RubenMattan) [inline src/solutions/common/cron/news/rss.py]: won't work for multiple news items
also please save url and guid if exists

File: src/solutions/common/cron/news/rss.py [hunk 2]
```diff
@@ -86,39 +92,37 @@ def _worker(rss_settings_key):
...
-                break
-
-        if dry_run:
-            run_in_transaction(trans_update_dry_run)
+                    to_put.append(new_item)
+            ndb.put_multi(to_put)
 
+        ndb.transaction(trans, xg=True)
+    if must_update_settings:
+        rss_settings.put()
```
Comment (RubenMattan) [inline src/solutions/common/cron/news/rss.py]: please save in transaction

File: src/solutions/common/cron/news/rss.py [hunk 3]
```diff
@@ -86,39 +92,37 @@ def _worker(rss_settings_key):
...
-                    db_item.dry_run = dry_run
-                    db_item.put()
+            url_key = SolutionRssScraperItem.create_key(service_user, service_identity, url)
+            keys.append(url_key)
+            if guid:
+                guid_key = SolutionRssScraperItem.create_key(service_user, service_identity, guid)
+                keys.append(guid_key)
+            items.append((title, url, guid, description_html, message))
 
+        scraper_items = {item.key.id(): item for item in ndb.get_multi(keys) if item}
```
Comment (RubenMattan) [inline src/solutions/common/cron/news/rss.py]: possible to get in transaction?

File: src/solutions/common/cron/news/rss.py (hunk 1 above)
Comment (lucasvanhalst) [reply src/solutions/common/cron/news/rss.py]: seems to work fine for me

**Comment by lucasvanhalst** [comment]
I don't see the point in adding transactions to any of this, this will only get executed once every hour so there will never be contention on these entities

File: src/solutions/common/cron/news/rss.py (hunk 1 above)
Comment (RubenMattan) [reply src/solutions/common/cron/news/rss.py]: how many items? what if they have 100 new items for example

**Pull request merged**